x-pack/plugins/lens/public/visualizations/datatable/components/columns.tsx

```diff
@@ -166,6 +167,52 @@ export const createGridColumns = (
       );
     }
 
+    cellActions.push(({ rowIndex, columnId, Component }: EuiDataGridColumnCellActionProps) => {
+      const { contentsIsDefined, cellContent } = getContentData({
+        rowIndex,
+        columnId,
+      });
+
+      const copyActionText = i18n.translate('xpack.lens.table.triggerAction.copyToClipboard', {
+        defaultMessage: 'Copy to clipboard',
+      });
+
+      if (!contentsIsDefined) {
+        return null;
+      }
+
+      const toastTitle = {
+        copyToClipboard: {
+          success: i18n.translate('xpack.lens.grid.copyValueToClipboard.toastTitle.success', {
+            defaultMessage: 'Copied to clipboard',
+          }),
+          error: i18n.translate('xpack.lens.grid.copyValueToClipboard.toastTitle.error', {
+            defaultMessage: 'Could not copy to clipboard',
+          }),
+        },
+      };
+
+      return (
+        <Component
+          aria-label={copyActionText}
+          data-test-subj="lensDatatableCopyToClipboard"
+          onClick={() => {
+            const { notifications } = pluginServices.getServices();
```

Comment:
We should not take the notifications service from the dashboard (it also adds 18KB to the dashboard bundle). We have this service in lens so you need to pass it as a prop to this function.